src/Drupal/Driver/Fields/Drupal8/LinkHandler.php

```diff
@@ -13,18 +13,34 @@ class LinkHandler extends AbstractHandler {
   public function expand($values) {
     $return = array();
     foreach ($values as $value) {
+      $uri = $value[1];
+
+      // If the uri has no scheme (and is not protocol relative) attempt to find
+      // a node with that label.
+      if (empty($uri_parts['scheme']) && strpos($uri, '//') !== 0) {
+        $entity_type_id = 'node';
+        $query = \Drupal::entityQuery($entity_type_id)->condition('title', $uri);
+        $entities = $query->execute();
+        if (!empty($entities)) {
+          $nid = array_shift($entities);
+          $uri = 'entity:' . $entity_type_id . '/' . $nid;
+        }
+      }
```

Comment:
I don't think this is the right way to approach this. The goal of DrupalDriver is to abstract the Drupal API, not the user interface.

When creating link fields through the API it is also not possible to supply a node title for the link field value and saving the entity like this. The resolving of the titles is a UI feature that is handled in the autocomplete. Behind the scenes the entity will be saved using the resolved link, not using the title that was originally entered in the UI.

 What would be interesting though is to support internal URIs, since these are also supported by the API. The `link_type` property on the link field controls whether or not internal links are accepted.